homeassistant/components/lastfm/sensor.py

```diff
@@ -60,11 +87,25 @@ class LastFmSensor(SensorEntity):
     _attr_attribution = "Data provided by Last.fm"
     _attr_icon = "mdi:radio-fm"
 
-    def __init__(self, user: User, lastfm_api: LastFMNetwork) -> None:
+    def __init__(self, api_key: str, username: str, entry_id: str) -> None:
         """Initialize the sensor."""
-        self._attr_unique_id = hashlib.sha256(user.name.encode("utf-8")).hexdigest()
-        self._attr_name = user.name
-        self._user = user
+        try:
+            self._user = LastFMNetwork(api_key=api_key).get_user(username)
```

Comment:
I think you should revert this and create the api connection in the setup as it was before.

Reply:
And pass the User object to the sensor? What if the user deleted their account? Are there any integrations that have good ways of handling individual entity failures?

Reply:
Probably use the `available` attribute and set to `false` if the user is not longer available (for some reason).
If it's the main user account you should raise `ConfigEntryAuthFailed` and setup a reauth step in config flow.

But I guess it would make sense to implement a data coordinator after this PR so you refresh all users in one go and the coordinator can then help out with the entities instead of each entity having an api connection.

Reply:
Uhm, we tried implementing a data update coordinator, but we reverted it at the end. @epenet will find the PR when I'm on my PC

Reply:
Perhaps it doesn't make sense with a coordinator but still think you should revert this change

Reply:
I need to correct myself here. I can do api.get_user("non existing user"), but then I'd get an User object which can't be used to make calls. In the setup of the config_flow (line 38) I call .get_playcount() to get pylast to throw the exception. So I think I can just create the sensor with an User object, but it wouldn't be a usable object, but then I'll move the validation part to the sensor